test/e2e/ocpbugs48050_test.go

```diff
@@ -0,0 +1,623 @@
+		if err := cl.List(context.TODO(), pods, client.InNamespace(namespace), client.MatchingLabels(labelSelector)); err != nil {
+			t.Logf("error listing pods in namespace %s with selector %v: %v", namespace, labelSelector, err)
+			return false, nil
+		}
+
+		for _, pod := range pods.Items {
+			isReady := false
+			for _, condition := range pod.Status.Conditions {
+				if condition.Type == corev1.PodReady && condition.Status == corev1.ConditionTrue {
+					isReady = true
+					break
+				}
+			}
+			if !isReady {
+				t.Logf("pod %s/%s is not ready", pod.Namespace, pod.Name)
+				return false, nil
+			}
+
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				if containerStatus.State.Terminated != nil {
+					t.Logf("pod %s/%s has a terminated container", pod.Namespace, pod.Name)
+					return false, nil
+				}
+				if containerStatus.State.Waiting != nil && containerStatus.RestartCount > 0 {
+					t.Logf("pod %s/%s is restarting (liveness probe failure)", pod.Namespace, pod.Name)
+					return false, nil
+				}
+			}
+		}
+
+		// All pods are ready and live.
+		return true, nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to observe readiness and liveness for pods in namespace %s with selector %v", namespace, labelSelector)
+	}
+
+	return nil
+}
+
+func singleTransferEncodingResponseCheck(resp *http.Response, err error) error {
+	if err != nil {
+		return fmt.Errorf("unexpected error: %v", err)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("unexpected status code: got %d, expected %d", resp.StatusCode, http.StatusOK)
+	}
+
+	_, err = io.Copy(io.Discard, resp.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read response body: %v", err)
+	}
+
+	return nil
+}
+
+func checkDuplicateTransferEncodingError(resp *http.Response, err error) error {
+	expectedErrorMsg := `net/http: HTTP/1.x transport connection broken: too many transfer encodings: ["chunked" "chunked"]`
+
+	if resp != nil {
+		resp.Body.Close()
+		return fmt.Errorf("expected response to be nil; got status: %s, code: %d", resp.Status, resp.StatusCode)
+	}
+
+	if err == nil {
+		return fmt.Errorf("expected an error, got nil")
+	}
+
+	if !strings.Contains(err.Error(), expectedErrorMsg) {
+		return fmt.Errorf("unmatched error: %v; expected message to contain %q", err, expectedErrorMsg)
+	}
+
+	return nil
+}
+
+func createOCPBUGS48050Service(namespace, name string) (*corev1.Service, error) {
+	service := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Annotations: map[string]string{
+				ingress.ServingCertSecretAnnotation:                  "serving-cert-" + namespace,
+				"service.beta.openshift.io/serving-cert-secret-name": "serving-cert-" + namespace,
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{"app": name},
+			Ports: []corev1.ServicePort{
+				{Name: "http", Port: 8080, TargetPort: intstr.FromInt(8080)},
```

Comment:
Linked to the question about the usage of http->https redirection. Do we bring some test value doing it? Or we can have just a single https port and simplify a little the configuration of the test env and server.

Reply:
I wanted to ensure we test the http path too. We have redirects on all the defined routes.